### PR TITLE
fix: hide parent operator button if form is disabled

### PR DESCRIPTION
**Pull request by marcellmueller**
Fix for #883

### CODE REVIEW

File: client/app/styles/rjsf/ArrayFieldTemplate.tsx [hunk 1]
```diff
@@ -67,8 +68,9 @@ const ArrayFieldTemplate = ({
           </div>
         );
       })}
-      {canAdd && (
+      {canAdd && !disabled && (
         <Button
+          disabled={disabled}
```
Comment (marcellmueller) [inline client/app/styles/rjsf/ArrayFieldTemplate.tsx:73]: This line is probably redundant but I added it just incase the button showed up somehow. Initially I had just disabled it using this though I though we should just hide it altogether if the form is disabled.

File: client/app/styles/rjsf/ArrayFieldTemplate.tsx [hunk 2]
```diff
@@ -67,8 +68,9 @@ const ArrayFieldTemplate = ({
           </div>
         );
       })}
-      {canAdd && (
+      {canAdd && !disabled && (
```
Comment (marcellmueller) [inline client/app/styles/rjsf/ArrayFieldTemplate.tsx:71]: Should we only hide this when disabled for internal users or is it fine to hide for everyone.

Comment (BCerki) [reply client/app/styles/rjsf/ArrayFieldTemplate.tsx:71]: Unknown, @nanyangpro ?

File: client/app/styles/rjsf/ArrayFieldTemplate.tsx (hunk 1 above)
Comment (BCerki) [reply client/app/styles/rjsf/ArrayFieldTemplate.tsx:73]: makes sense to me

**Review by BCerki — COMMENTED** [review]
Nice work! I looked at the form as an IDIR user and couldn't see the button, and the button was still available as an internal user while editing. I don't know the answer to your question; I tagged Nan. I also committed a fixture to add a parent operator because that made manual testing easier; feel free to edit it if you want

File: client/app/styles/rjsf/ArrayFieldTemplate.tsx (hunk 2 above)
Comment (BCerki) [reply client/app/styles/rjsf/ArrayFieldTemplate.tsx:71]: Talked with Nan, will hide (rather than disable) button for everyone

**Review by BCerki — APPROVED** [review]

**Comment by nanyangpro** [comment]
Thanks for confirming, @BCerki!
- As discussed, in read-only views for the form, either disabling / graying out OR hiding the button is fine. Ideally we hide it.
- For future enhancement, we should consider having a generic look and feel and behaviour for forms when read-only.

**Pull request merged**